jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/BufferUtil.java

```diff
@@ -595,6 +596,39 @@ public static void readFrom(Path path, ByteBuffer buffer) throws IOException
         }
     }
 
+    /**
+     * Read content from a {@link ReadableByteChannel} into a buffer.
+     * This may spin if {@link ReadableByteChannel#read(ByteBuffer)} returns 0 in which case this
+     * will call {@link Thread#onSpinWait()}.
+     *
+     * @param readableByteChannel the channel to read from.
+     * @param contentLengthValue the length of the content to read.
+     * @param byteBuffer the buffer to read into.
+     * @throws IOException if an I/O error occurs.
+     */
+    public static void readFrom(ReadableByteChannel readableByteChannel, long contentLengthValue, ByteBuffer byteBuffer) throws IOException
+    {
+        int pos = BufferUtil.flipToFill(byteBuffer);
+        try
+        {
+            int totalRead = 0;
```

Comment:
Should be a `long`.

Reply:
I don't think so as it is size limited by the ByteBuffer, which can only be int sized.

Reply:
done